packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -3917,6 +3918,44 @@ class Stack extends MultiChildRenderObjectWidget {
   }
 }
 
+class IndexedStack extends StatelessWidget {
+  const IndexedStack({
+    super.key,
+    this.alignment = AlignmentDirectional.topStart,
+    this.textDirection,
+    this.clipBehavior = Clip.hardEdge,
+    this.sizing = StackFit.loose,
+    this.index = 0,
+    this.children = const <Widget>[],
+  });
+
+  final AlignmentGeometry alignment;
+  final TextDirection? textDirection;
+  final Clip clipBehavior;
+  final StackFit sizing;
+  final int? index;
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> wrappedChildren = List<Widget>.generate(children.length, (int i) {
+      return Visibility(
```

Comment:
To maintain the documented properties of the IndexedStack ("The stack is always as big as the largest child.") this must set `maintainSize` to true.